sdk/src/binary.rs

```diff
@@ -371,35 +350,45 @@ impl<const N: u32> Ord for ArrayBinary<N> {
     }
 }
 
-impl<const N: u32> FixedLengthBinary for ArrayBinary<N> {
-    #[inline(always)]
-    fn put(&mut self, i: u32, v: u8) {
-        self.0.put(i, v);
+impl<const N: u32> Borrow<Binary> for ArrayBinary<N> {
+    fn borrow(&self) -> &Binary {
+        &self.0
     }
+}
 
-    #[inline(always)]
-    fn get(&self, i: u32) -> u8 {
-        self.0.get(i)
+impl<const N: u32> BorrowMut<Binary> for ArrayBinary<N> {
+    fn borrow_mut(&mut self) -> &mut Binary {
+        &mut self.0
     }
+}
 
-    #[inline(always)]
-    fn is_empty(&self) -> bool {
-        false
+impl<const N: u32> Borrow<Binary> for &ArrayBinary<N> {
+    fn borrow(&self) -> &Binary {
+        &self.0
     }
+}
 
-    #[inline(always)]
-    fn len(&self) -> u32 {
-        N
+impl<const N: u32> Borrow<Binary> for &mut ArrayBinary<N> {
+    fn borrow(&self) -> &Binary {
+        &self.0
     }
+}
 
-    #[inline(always)]
-    fn front(&self) -> u8 {
-        self.0.front()
+impl<const N: u32> BorrowMut<Binary> for &mut ArrayBinary<N> {
```

Comment:
Isn't this unsafe? I could change the length of an `ArrayBinary` with this.

Reply:
You're right. I'll remove the BorrowMut and AsMut.